lightkurve/mast.py

```diff
@@ -61,86 +69,242 @@ def search_kepler_products(target):
             target_name = 'ktwo{:09d}'.format(target)
         else:
             raise ValueError("Not in the Kepler KIC or EPIC ID range")
-        obs = Observations.query_criteria(target_name=target_name,
+        obs = Observations.query_criteria(objectname=target_name,
+                                          radius='{} arcsec'.format(radius),
                                           project=["Kepler", "K2"])
+        #Remove KeplerFFI data set.
+        obs = obs[(obs['obs_collection']=='Kepler') | (obs['obs_collection']=='K2')]
+        obs.sort('distance')
     except ValueError:
         # If `target` did not look like a KIC or EPIC ID, then we let MAST
         # resolve the target name to a sky position.
         try:
             obs = Observations.query_criteria(objectname=target,
-                                              radius='1 arcsec',
+                                              radius='{} arcsec'.format(radius),
                                               project=["Kepler", "K2"])
+            #Remove KeplerFFI data set.
+            obs = obs[(obs['obs_collection']=='Kepler') | (obs['obs_collection']=='K2')]
+            obs.sort('distance')
         except ResolverError as exc:
             raise ArchiveError(exc)
-    return Observations.get_product_list(obs)
+
+    #Make sure the final table is in DISTANCE order.
+    obsids = np.asarray(obs['obsid'])
+    t = (Observations.get_product_list(obs))
+    l = [np.where(t['parent_obsid'] == o)[0]  for o in obsids]
+    l = [item for sublist in l for item in sublist]
+    t = t[l]
+
+    return t
 
 
-def search_kepler_tpf_products(target, cadence='long', quarter=None,
-                               campaign=None):
-    """Returns a table of Kepler or K2 Target Pixel Files for a given target.
+def search_kepler_products(target, filetype='Target Pixel', cadence='long', quarter=None,
+                               campaign=None, radius=1):
+    """Returns a table of Kepler or K2 Target Pixel Files or Lightcurve Files
+     for a given target.
 
     Parameters
     ----------
     cadence: 'short' or 'long'
         Specify short (1-min) or long (30-min) cadence data.
+    filetype : 'Target Pixel' or 'Lightcurve'
```

Comment:
Let's use `TargetPixelFile` and `LightCurveFile` as the permissible options to be consistent with the terminology elsewhere in lightkurve.  (Unless this makes the code significantly more complicated.)